app/controllers/posts_controller.rb

```diff
@@ -7,10 +7,18 @@ def index
 
   def show
     @post = Post.find(params[:id])
-    @restaurant = Restaurant.find(params[:restaurant_id])
+    restaurant = @post.restaurant_id
+    @restaurant = Restaurant.find(restaurant)
```

Comment:
how about 
```
@restaurant = @post.restaurant
```

this should work if you have `belongs_to :restaurant` in your post model

Reply:
I didn't have that association set up, which is why @post.restaurant wasn't accessible. Ill make that correction.